ambari-server/src/main/java/org/apache/ambari/server/topology/PersistedStateImpl.java

```diff
@@ -311,7 +286,30 @@ private TopologyHostRequestEntity toEntity(HostRequest request, TopologyLogicalR
     return entity;
   }
 
-  static final class ReplayedTopologyRequest implements TopologyRequest {
+  // FIXME hard-code
+  static final class ReplayedProvisionRequest extends ReplayedTopologyRequest implements ProvisionRequest {
+
+    ReplayedProvisionRequest(TopologyRequestEntity entity, BlueprintFactory blueprintFactory) {
+      super(entity, blueprintFactory);
+    }
+
+    @Override
+    public ConfigRecommendationStrategy getConfigRecommendationStrategy() {
+      return ConfigRecommendationStrategy.NEVER_APPLY;
+    }
+
+    @Override
+    public String getDefaultPassword() {
+      return null;
+    }
+
+    @Override
+    public SecurityConfiguration getSecurityConfiguration() {
+      return SecurityConfiguration.NONE;
+    }
+  }
+
```

Comment:
What is the purpose of the hard-coded overrides?

Reply:
These are not overrides, but necessary implementations of interface methods.  I'm not sure they are used at all during request replay.

Changed the hard-coded values to `null`.